src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx

```diff
@@ -86,6 +86,8 @@ const BarplotConfig = t.intersection([
     overlayVariable: VariableDescriptor,
     facetVariable: VariableDescriptor,
     showMissingness: t.boolean,
+    // new config props to store filters at fullscreen mode
+    currentPlotFilters: t.unknown,
```

Comment:
Instead of adding `currentPlotFilters` to each viz config, you could make `currentPlotFilters` a top-level property of the `VisualizationDescriptor`:

```typescript
/**
 * Metadata for the visualization object stored in user's analysis
 */
export type VisualizationDescriptor = TypeOf<typeof VisualizationDescriptor>;
export const VisualizationDescriptor = intersection([
  type({
    type: string,
    configuration: unknown,
  }),
  partial({
    thumbnail: string,
    currentPlotFilters: t.array(Filter)
  }),
]);
```

Reply:
@jtlong3rd as I said above, I also defined the `currentPlotFilters` props in there: actually there is no generic `t` in there so just used `array(Filter)`